### DISCUSSION TITLE
Contributions on separation processes

### DISCUSSION THREAD

**Discussion by jAniceto** [wigging/chemics#17] [General]
Hi. 

I see Chemics is mostly focused on fluidization processes. Would there be interested in adding functions in other chemical engineering areas? 

I am a researcher and have worked with adsorption, chromatography and ion exchange. I would be willing to contribute in these areas if it is of interest.

**Comment by wigging** [comment]
Yes, functions related to other chemical engineering areas are welcome in Chemics. I have been doing a lot of fluidization research so that is why the package is mostly focused on fluidization functions. But I would definitely like to see others contribute to the package to make it applicable to a broader range of applications. I'm not familiar with liquid chromatography or ion exchange. What would be some examples of functions that you could contribute in that area?

**Reply by jAniceto** [reply]
My idea is to start with adsorption and chromatography. I could add:
- adsorption isotherm models such as Langmuir, Freundlich and several others.
- functions to fit these models to experimental data
- chromatographic models (for simulating several processes)

Other stuff that is more general but needed in this field like:
- pressure drop calculations (Ergun equation for instance)
- correlations for the calculation of several transport properties such as diffusivity coefficients, axial dispersion, mass transfer coefficients, etc.
- relevant dimensionless numbers like Reynolds, Pecker (I believe you already have some of this).

On the field of ion exchange I can add models for the calculation of activity coefficients in solid and liquid phases like Debye-Huckel, Pitzer, Wilson, NRTL and UNIQUAC.

I have a lot of these already coded, though mostly in Matlab. I have intended to collect it in a Python package for some time. When I saw Chemics I thought I could add to it rather than start a new package.

**Reply by wigging** [reply]
All of this would be welcome in the Chemics package. See the [contributing guidelines](https://github.com/chemics/chemics/blob/main/CONTRIBUTING.md) for some suggestions on how to contribute to the package. Thank you for your interest in the package and I'm looking forward to your contributions.

**Reply by jAniceto** [reply]
Great! I think I can have something ready in a couple of weeks.

**Comment by wigging** [comment]
Have you contributed to other Python projects on GitHub or will this be your first one?

**Reply by jAniceto** [reply]
Will be the first time contributing to an open-source project. Hope it's ok. I have done several personal/work Python (and Matlab) projects thought.

**Reply by wigging** [reply]
That is fine. I would suggest doing a small Pull Request for one or two functions as your first contribution. Once you get familiar with the process then you can make larger commits. If you have any questions about the process, then please ask in the Q&A section of this Discussions forum.

**Reply by jAniceto** [reply]
Will do. Thanks.

**Comment by wigging** [comment]
Chemics v21.5 was released and includes your contributions. Thank you for adding to the package.

**Reply by jAniceto** [reply]
Hey! Thanks. I've been a bit busy lately but I already have more stuff lined up. Only needs a bit more testing.

**Reply by wigging** [reply]
No problem. Just wanted to let you know about the release.
